client/web/src/onboarding-tour/utils.tsx

```diff
@@ -0,0 +1,23 @@
+export const buildURIMarkers = (href: string, stepId: string): string => {
+    const isRelative = !isAbsoluteURL(href)
+
+    try {
+        const url = new URL(href, isRelative ? `${location.protocol}//${location.host}` : undefined)
+        url.searchParams.set('tour', 'true')
+        url.searchParams.set('stepId', stepId)
+        return isRelative ? url.toString().slice(url.origin.length) : url.toString()
+    } catch {
+        return '#'
+    }
+}
+
+export const parseURIMarkers = (searchParameters: string): { isTour: boolean; stepId: string | null } => {
+    const parameters = new URLSearchParams(searchParameters)
+    const isTour = parameters.has('tour')
+    const stepId = parameters.get('stepId')
+    return { isTour, stepId }
+}
+
+const isAbsoluteURL = (url: string): boolean => url.startsWith('http')
```

Comment:
nit: Something like this might be more resilient (please double-check my regex):

```suggestion
const isAbsoluteURL = (url: string): boolean => url.startsWith(/https?:\/\//i')
```

Reply:
Thanks, updated! 
P.S: seems like startsWith doesn't accept ReGex, so used `/^https?:\/\//i.test(url)` instead